### ISSUE TITLE
 gmod_wire_pod & AD2 issues

### ISSUE THREAD

**Issue by TheFreezebug** [wiremod/wire#1003]
Sorry for a long ranty post, but here goes. 

I encounter a problem when pasting older dupes that requires me to manually whitelist specific entities. I think (think) part of it has to do with the BuildDupeInfo. 

Example:

Regardless of wire_pod_controller being aliased as gmod_wire_adv_pod, AD2 thinks this is is for some reason an NPC.  I don't know whether or not to put this in ad2 github or here, but since this is where I encounter my problem. I'll put it here.

The BuildDupeInfo for wire_pod_controller is (I think) outdated.

    --Duplicator support to save pod link (TAD2020)
    function ENT:BuildDupeInfo()
    	local info = self.BaseClass.BuildDupeInfo(self) or {}
    	if (self:HasPod() and !self.RC) then
    		info.pod = self.Pod:EntIndex()
    	end
    	return info
    end

This is a function, whereas AD2 asks for a TABLE

    if( game.SinglePlayer() or AdvDupe2.duplicator.WhiteList[EntTable.Class]  or     (EntTable.BuildDupeInfo.IsNPC and (tobool(GetConVarString("AdvDupe2_AllowNPCPasting")) and     string.sub(EntTable.Class, 1, 4)=="npc_")))then

(I don't actually know how BuildDupeInfo works so it could be just fine and me not know it)

Next: 

I saw this at the end of gmod_wire_pod.lua

    scripted_ents.Alias("gmod_wire_adv_pod", "gmod_wire_pod")
And it didn't make sense to me until I realized scripted_ents.Alias doesn't actually alias the entity until one tick has passed, so this is why ad2 doesn't catch it and immediately whitelist it as a scripted entity. Another part of your problem with this is the fact of which you're calling this function before the entity is fully created. Putting this on the last line of gmod_wire_pod.lua means the entity you're trying to alias doesn't even exist yet, and in fact won't exist until AFTER 

    scripted_ents.Alias("gmod_wire_adv_pod", "gmod_wire_pod")

is called

Long story short, it'd be nice if this could be remedied somehow, whether its manually inserting it into the whitelist or using some other method.

**Comment by thegrb93** [comment]
What is telling you that it is an npc?

**Comment by TheFreezebug** [comment]
Nah, it's telling me ENTITY CLASS BLACKLISTED: gmod_wire_adv_pod

**Comment by Divran** [comment]
`scripted_ents.Alias("gmod_wire_adv_pod", "gmod_wire_pod")` is called once, when the first launches. It's not called every time that entity is created.

**Comment by TheFreezebug** [comment]
@Divran The error "ENTITY CLASS BLACKLISTED" is serverside. 
Your statement is right, but garrysmod registers the actual gmod_wire_pod entity AFTER scripted_ents.Alias("gmod_wire_adv_pod", "gmod_wire_pod"). When you alias it, the entity doesn't exist yet.

**Comment by Divran** [comment]
Checked adv dupe 2 logs. This bug was fixed over a year ago
https://github.com/wiremod/advdupe2/pull/5
